image-processing/core/preprocessing.py

```diff
@@ -47,3 +47,64 @@ def remove_shadows(image: str | cv2.typing.MatLike, output: Optional[str] = None
         os.remove(output)
 
     return res
+
+
+def increase_contrast(image: str | np.ndarray, output: Optional[str] = None, alpha: float = 1.5, beta: float = 0) -> np.ndarray:
+
+    image, tif_meta = _load_image(image)
+
+    # Apply contrast adjustment using the formula: output = alpha * input + beta
+    res = cv2.convertScaleAbs(image, alpha=alpha, beta=beta)
+
+    if output is not None:
+        _save_image(res, output, tif_meta)
+
+    return res
+
+
+def increase_saturation(image, output=None, factor=1.5):
+
+    image, tif_meta = _load_image(image)
+
+    # Convert the image from BGR to HSV
+    hsv_image = cv2.cvtColor(image, cv2.COLOR_BGR2HSV)
+
+    # Increase the saturation channel
+    hsv_image[:, :, 1] = np.clip(
+        hsv_image[:, :, 1] * factor, 0, 255).astype(np.uint8)
+
+    # Convert the image back to BGR
+    res = cv2.cvtColor(hsv_image, cv2.COLOR_HSV2BGR)
+
+    if output is not None:
+        _save_image(res, output, tif_meta)
+
+    return res
+
+
+
+def _load_image(image: str | np.ndarray):
+    meta = None
+    if isinstance(image, str):
+        if image.endswith(".tif"):
+            with rasterio.open(image) as src:
+                meta = src.meta.copy()
+                # Read with Rasterio and transpose to OpenCV format:
+                # from (bands, height, width) to (height, width, bands)
+                image = src.read().transpose(1, 2, 0)
+        else:
+            image = cv2.imread(image)
+
+    return image, meta
+
+def _save_image(res: np.ndarray, output: str, tif_meta: dict):
```

Comment:
Same here missing return type hint for None and tif_meta should be an Optional[dict]

Reply:
erledigt